tools/app/app/components/StateSelector.tsx

```diff
@@ -16,64 +24,50 @@ const StateSelector: React.FC<StateSelectorProps> = ({
   workflow,
   onTransition,
 }) => {
-  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
-  const open = Boolean(anchorEl)
-
-  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
-    setAnchorEl(event.currentTarget)
-  }
-
-  const handleClose = () => {
-    setAnchorEl(null)
-  }
-
   const handleMenuItemClick = (transition: WorkflowTransition) => {
     onTransition(transition)
-    handleClose()
   }
 
-  if (
-    currentState == null ||
-    workflow == null ||
-    !workflow.states.find((state) => state.name == currentState.name)
-  )
-    return null
+  const { t } = useTranslation()
 
-  const availableTransitions = workflow?.transitions.filter(
-    (transition) =>
-      (transition.fromState.includes(currentState.name) ||
-        transition.fromState.includes('*')) &&
-      transition.toState !== currentState.name
-  )
+  const availableTransitions = useMemo(() => {
+    if (
+      !currentState ||
+      !workflow ||
+      !workflow.states.find((state) => state.name == currentState.name)
+    ) {
+      return null
+    }
+
+    return workflow?.transitions.filter(
+      (transition) =>
+        (transition.fromState.includes(currentState.name) ||
+          transition.fromState.includes('*')) &&
+        transition.toState !== currentState.name
+    )
+  }, [currentState, workflow])
+
+  if (!availableTransitions || !currentState) return null
 
   return (
-    <div>
-      <Button
+    <Dropdown>
+      <MenuButton
+        size="sm"
         disabled={availableTransitions.length == 0}
         variant="soft"
-        onClick={handleClick}
-        color="neutral"
-        size="sm"
         startDecorator={
           <FiberManualRecordIcon
             style={{ color: getStateColor(currentState) }}
           />
         }
-        sx={{
-          pl: 2,
-          pr: 2,
-          marginLeft: 1,
-        }}
-      >
-        <strong>Status: {currentState.name}</strong>
-      </Button>
-      <Menu
-        id="status-menu"
-        anchorEl={anchorEl}
-        keepMounted
-        open={open}
-        onClose={handleClose}
       >
+        <Typography fontWeight={600}>
+          {t('stateSelector.status', {
+            state: currentState.name,
+          })}
+        </Typography>
+      </MenuButton>
+      <Menu>
         {availableTransitions.map((transition) => (
           <MenuItem
             key={transition.name}
```

Comment:
Maybe onClick could just call onTransition directly?

Reply:
Maybe it could even call redux directly, but I updated it to use onTransition for now